packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -147,14 +142,26 @@ export const useCheckoutPage = props => {
     ]);
 
     useEffect(() => {
-        if (cartId) {
+        const placedOrder = !!placeOrderData;
+
+        /**
+         * Get checkout details everytime `cartId` changes
+         * but not if the place order mutation is called
+         * and `cartId` changes because of that.This is
+         * because after the place order mutation is completed
+         * old `cartId` will be purged and a new one will be
+         * created. we wont need checkout details for that
+         * `cartId`, since the order has been placed and the UI
+         * will be in the order confirmation step.
+         */
+        if (cartId && !placedOrder) {
```

Comment:
Tommy had a really good idea in his auth shipping PR. Instead of conditions in effects to invoke lazy queries we can just use the `skip` property. 

```
https://github.com/magento/pwa-studio/pull/2380/files#r430714912
11:09
I wonder - could you use this instead of the effect conditional?
11:10
If you replace the checkout page query with this we won't need the effect to call the lazy query anyways.
const {
    data: checkoutData
} = useQuery(getCheckoutDetails, {
    skip: !cartId,
    variables: {
        cartId
    }
});
```

Reply:
I am fine doing it either way. I like this style as well. Just a small change though, the skip variable value will be `cartId && !!placeOrderData`.

> However after I mention this I realize that it is currently possible for a user to log out/log in while in cart/checkout which would require we re-fetch the data. I don't know if this would re-fetch in those cases.

Ideally speaking it should. Technically speaking, it should since the `cartId` changes once you log in or out.

Reply:
I have tried this and it worked like a charm. When I logged in and out, I could see that the query was being fired.

What do you guys think? Should we stick with this approach?

Reply:
Did it work when logging in and logging out? The checkout page reflected the user state ie showed authed user cart after logging in and showed empty cart after logging out?

Reply:
Yeah, it did. Because when we log we use the `cartId` associated with the user and the query would pick that up and do its magic. Once the user logs out, a new cartId is created so the query checks and realizes there are not cart items on it. That's my understanding.

Reply:
I'm 100% in support of this change, and was about to implement something similar across all lazy queries in scope of Checkout. I've tested that logging in and out triggers a re-fetch as expected, and have not found any problems so far. Please proceed with this change, but try to keep your scope tight; nearly everything in here is a conflict with what I have in flight with Auth'd Shipping Info.

Reply:
Quick point, I have tried both the use cases we are discussing here, `useEffect` and `skip`, unfortunately, they are a bit diff. [I have documented it in this video](https://adobe-my.sharepoint.com/personal/annavara_adobe_com/_layouts/15/guestaccess.aspx?guestaccesstoken=052Ws%2BmIduH8kYSiDI2V3buQ%2BMMkwRi2P8pTA%2Bb9IJE%3D&docid=2_088750a8d85aa4b56a605d54aea8537ef&rev=1&e=q74ebI).

TLDR, `useEffects` does not get invoked if the value of the variable changes to the same value like, `123` -> `123` but the `skip` variable does not recognize that. It just sees that the value has changed but not the actual value. It does not compare the new value to the old value. When ever the value changes, if that value passes the conditional, the `useQuery` will perform its action.

Reply:
Found the link of code that is causing the loading screen flash.

The loading screen is controlled by the boolean `isLoading` in the `useCheckoutPage` talon. This is how the `isLoading` is defined:

```es6
isLoading: !checkoutCalled || (checkoutCalled && checkoutLoading)
```

It reads as:
1. if the checkout details query is not called, show the loading screen
2. if it is called and is loading, show the loading screen

If you pay attention to the 2 screenshots below, in case of `useEffect`, `checkoutCalled` is not set to `true` till it is called, ie. the `useEffect` is executed.

In case of `useQuery`, it is called when the component is rendered, hence `checkoutCalled` is set to `true` even on the first render but it is not loading yet. Hence the `isLoading` boolean is false first, then when `checkoutLoading` is actually set to true, `isLoading` is set to true.

I hope this explanation of mine is understandable 🤣 

`useEffect`

![image](https://user-images.githubusercontent.com/35203638/83061843-40390680-a023-11ea-9de6-8b7008a0d04c.png)

`skip`

![image](https://user-images.githubusercontent.com/35203638/83061537-c86adc00-a022-11ea-86e0-e53165b30c0c.png)

Reply:
Maybe we should revisit the apollo docs for loading states: https://www.apollographql.com/docs/react/data/queries/#inspecting-loading-states

Reply:
Pstephen the savior. Using the `networkStatus` variable from the `useQuery` worked dude.

Reply:
Should I be happy that the tests failed to catch the change or sad that I have to update them? 🤔